### PR TITLE
Improve diagnostics extension command

### PR DESCRIPTION
**Pull request by yantang-msft**
This code change has gone through internal code review. Reviewer: @bfung, @rapandit

And the changes including:
1. Allow user specify storage account name, key, endpoint instead of a mandatory StorageContext parameter
2. Allow user specify an array of configuration for SetAzureServiceDiagnosticsExtension commands, so we can apply different configurations within one call.
3. Bug fix
4. Add scenario tests

### CODE REVIEW

**Comment by bfung** [comment]
I don't believe I'm the _bfung_ you're looking for, there are some simple fixes I could suggest, but perhaps they've been caught by the build? ;)

**Comment by yantang-msft** [comment]
@bfung, didn't noticed the alias points to you by mistake.
What kind of simple fix would you suggest? The build failed for scenario test, while it succeeded on my local machine. I'm looking into it.

File: src/ServiceManagement/Compute/Commands.ServiceManagement/Common/DiagnosticsHelper.cs [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+            }
+        }
+
+        public static Hashtable GetPublicDiagnosticsConfiguration(string config, string storageAccountName)
+        {
+            // find the <WadCfg> element and extract it
+            int wadCfgBeginIndex = config.IndexOf("<WadCfg>");
+            if (wadCfgBeginIndex == -1)
+            {
+                throw new ArgumentException("Cannot find the WadCfg element in the config.");
```
Comment (hovsepm) [inline src/ServiceManagement/Compute/Commands.ServiceManagement/Common/DiagnosticsHelper.cs]: Please add all strings for the exceptions to the resources and reference it from there.

Comment (yantang-msft) [inline src/ServiceManagement/Compute/Commands.ServiceManagement/Common/DiagnosticsHelper.cs]: > Please add all text strings for the exceptions to the resources and reference it from there.

Thanks! Done.

**Pull request merged**